test/test_logging.py

```diff
@@ -204,3 +204,77 @@ def function_that_raises_and_adds_to_stack_trace():
                 }
             ],
         }
+
+
+class TestCompositeLogger:
+    @pytest.mark.parametrize(
+        "method, expected_level",
+        [
+            ("debug", "debug"),
+            ("info", "info"),
+            ("warning", "warning"),
+            ("error", "error"),
+            ("fatal", "critical"),
+            ("exception", "exception"),
+            ("critical", "critical"),
+        ],
+    )
+    def test_debug(self, capfd: LogCaptureFixture, method: str, expected_level: str) -> None:
+        haystack_logging.configure_logging(use_json=True)
+
+        logger = haystack_logging.getLogger(__name__)
+        logger.logger.setLevel(logging.DEBUG)
+
+        getattr(logger, method)("Hello, structured logging!", key1="value1", key2="value2")
+
+        output = capfd.readouterr().err
+        parsed_output = json.loads(output)  # should not raise an error
+
+        assert parsed_output == {
+            "event": "Hello, structured logging!",
+            "key1": "value1",
+            "key2": "value2",
+            "level": expected_level,
+            "timestamp": ANY,
+        }
+
+    def test_log(self, capfd: LogCaptureFixture) -> None:
+        haystack_logging.configure_logging(use_json=True)
+
+        logger = haystack_logging.getLogger(__name__)
+        logger.logger.setLevel(logging.DEBUG)
+
+        logger.log(logging.DEBUG, "Hello, structured logging!", key1="value1", key2="value2")
+
+        output = capfd.readouterr().err
+        parsed_output = json.loads(output)
+
+        assert parsed_output == {
+            "event": "Hello, structured logging!",
+            "key1": "value1",
+            "key2": "value2",
+            "level": "debug",
+            "timestamp": ANY,
+        }
+
+    def test_that_haystack_logger_is_used(self) -> None:
```

Comment:
@masci What do you think - is this something we want to force? Reasoning is that I'd like that people think more consciously about adding extras to their code to make the logs more informative:

Rich logs also have the advantage that would we could simply add all of that  as tags to the currently traced span.

1. Technically we could even enforce to add a `kwarg` whenever we log a positional message -> ensure that the positional argument is captured as kwarg
2. Add all keys from logging as tags to tracing

Reply:
Honestly I like the idea, logging has traditionally been improvised across the codebase, and this would bring in ultimate consistency - the tradeoff is advantageous.